### ISSUE TITLE
Support postgres `all` function

### ISSUE THREAD

**Issue by spiffytech** [kysely-org/kysely#1303]
The `all` API is just like `any`'s. Whereas `any` is used for 'is this value present', 'all' is often used for ['is this value _not_ present'](https://stackoverflow.com/a/34627688), so it feels appropriate for both functions to be supported.

**Comment by koskimas** [comment]
When is `all` useful? It just checks that the array (or other expression) is full of one single value?

**Comment by koskimas** [comment]
Oh, right. You can do "not in" using `foo != ALL(array)`.

**Comment by spiffytech** [comment]
Yep: `any` and `all` are often used for semantically similar work to `in` and `not in`, but for working with arrays.
